src/FileStorageServiceProvider.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Yiisoft\Yii\Filesystem;
+
+use League\Flysystem\FilesystemAdapter;
+use Yiisoft\Di\Container;
+use Yiisoft\Di\Support\ServiceProvider;
+use Yiisoft\Factory\Factory;
+
+final class FileStorageServiceProvider extends ServiceProvider
+{
+    public function register(Container $container): void
+    {
+        $factory = new Factory();
```

Comment:
May use DI?

Reply:
In what way?

Reply:
To get the `Factory` from `$container` instead of making new instance

Reply:
It's a wrong way, it will be instantiated with the main app container.